mobile/src/main/java/org/openhab/habdroid/core/OnUpdateBroadcastReceiver.java

```diff
@@ -0,0 +1,49 @@
+package org.openhab.habdroid.core;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import org.openhab.habdroid.ui.OpenHABMainActivity;
+import org.openhab.habdroid.util.Constants;
+
+public class OnUpdateBroadcastReceiver extends BroadcastReceiver {
+    private static final String TAG = OpenHABMainActivity.class.getSimpleName();
+
+    private static final String UPDATE_LOCAL_CREDENTIALS = "1.8.0.8";
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (getVersionNumber(context.getPackageManager(), context.getPackageName())
+                .equals(UPDATE_LOCAL_CREDENTIALS)) {
```

Comment:
We'd assign a number as pref version, which is counted up with each pref change that needs conversion. Conversion code reads the number, compares it to the known latest version number and does conversion based on the difference. We wouldn't need to remove any code snippet, as the conversion just wouldn't do anything if the version matches.
And no need to do the conversion in the activity, doing it in this receiver is fine (props for discovering that intent action!)

Reply:
But what if the version of the app wasn't updated so far (but just installed the first time) and then updated? It will need some default value (let's say 0), which would trigger all conversions, which would be incorrect, wouldn't it?